src/test/java/app/bpartners/api/unit/service/CustomerServiceTest.java

```diff
@@ -0,0 +1,125 @@
+import app.bpartners.api.repository.CustomerRepository;
+import app.bpartners.api.repository.ban.BanApi;
+import app.bpartners.api.repository.ban.model.GeoPosition;
+import app.bpartners.api.service.CustomerService;
+import app.bpartners.api.service.utils.GeoUtils;
+import java.util.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CustomerServiceTest {
+  CustomerRepository repositoryMock;
+  BanApi banApiMock;
+  CustomerRestMapper restMapperMock;
+  EventProducer eventProducerMock;
+  EventConf eventConfMock;
+  SesConf sesConfMock;
+  CustomerService subject;
+
+  @BeforeEach
+  void setUp() {
+    repositoryMock = mock(CustomerRepository.class);
+    banApiMock = mock(BanApi.class);
+    subject =
+        new CustomerService(
+            repositoryMock,
+            restMapperMock,
+            eventProducerMock,
+            eventConfMock,
+            sesConfMock,
+            banApiMock);
+  }
+
+  Customer.CustomerBuilder customerBuilder() {
+    return Customer.builder().id(JOE_DOE_ID);
+  }
+
+  @Test
+  void testUpdateCustomersLocationAddressTooShort() {
+    var customer = customerBuilder().address("12").build();
+
+    when(repositoryMock.findWhereLatitudeOrLongitudeIsNull()).thenReturn(List.of(customer));
+
+    subject.updateCustomersLocation();
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationAddressTooLong() {
+    var customer = customerBuilder().address("A".repeat(201)).build();
+
+    when(repositoryMock.findWhereLatitudeOrLongitudeIsNull()).thenReturn(List.of(customer));
+
+    subject.updateCustomersLocation();
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationAddressNotFound() {
+    var customer = customerBuilder().address("123 Street").build();
+
+    when(repositoryMock.findWhereLatitudeOrLongitudeIsNull()).thenReturn(List.of(customer));
+    when(banApiMock.search("123 Street")).thenReturn(null);
+
+    subject.updateCustomersLocation();
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationExceptionHandling() {
+    var customer = customerBuilder().address("123 Street").build();
+
+    when(repositoryMock.findWhereLatitudeOrLongitudeIsNull()).thenReturn(List.of(customer));
+    when(banApiMock.search(any())).thenThrow(new BadRequestException("Bad Request"));
+
+    subject.updateCustomersLocation();
+    verify(repositoryMock, never()).save(any(Customer.class));
+  }
+
+  @Test
+  void testUpdateCustomersLocationSuccessfulUpdate() {
+    var geoPosition = mock(GeoPosition.class);
+    var coordinates = mock(GeoUtils.Coordinate.class);
+    var customer = mock(Customer.class);
+
+    when(customer.getId()).thenReturn("1");
+    when(customer.getFullAddress()).thenReturn("123 Main St, Springfield");
+    when(customer.getAddress()).thenReturn("123 Main St, Springfield");
```

Comment:
Do not mock customer, create a Customer object instead and mock the method to return it instead